apps/condo/domains/acquiring/schema/CreatePaymentByLinkService.js

```diff
@@ -161,97 +132,92 @@ const CreatePaymentByLinkService = new GQLCustomSchema('CreatePaymentByLinkServi
-                            id: acquiringContexts[0].id,
-                        },
-                    })
 
-                    multiPaymentId = id
-                } else if (lastBillingReceipt.period === period) {
-                    // if period matches we use found receipt to create MultiPayment
-                    await validateReceiptDuplicates(context, PersAcc, lastBillingReceipt.period, organizationsIds, PersonalAcc)
-                    const { multiPaymentId: id } = await registerMultiPaymentForOneReceipt(context, {
-                        dv, sender,
-                        receipt: { id: lastBillingReceipt.id },
-                        acquiringIntegrationContext: { id: contextsMap[get(lastBillingReceipt, ['context', 'id'])] },
-                    })
-                    multiPaymentId = id
-                } else if (lastBillingReceipt.period > period) {
-                    // we have a newer receipt at our side - let's pay for newer one
-                    await validateReceiptDuplicates(context, PersAcc, lastBillingReceipt.period, organizationsIds, PersonalAcc)
+                const payForLastBillingReceipt = async (lastBillingReceipt) => {
+                    if (await isReceiptPaid(context, PersAcc, lastBillingReceipt.period, organizationsIds, PersonalAcc)) {
+                        throw new GQLError(ERRORS.RECEIPT_ALREADY_PAID, context)
+                    }
                     const { multiPaymentId: id } = await registerMultiPaymentForOneReceipt(context, {
                         dv, sender,
                         receipt: { id: lastBillingReceipt.id },
-                        acquiringIntegrationContext: { id: contextsMap[get(lastBillingReceipt, ['context', 'id'])] },
+                        acquiringIntegrationContext: { id: contextsMap[get(lastBillingReceipt, ['context'])] },
                     })
                     multiPaymentId = id
-                } else {
-                    // the last receipt is older than the scanned one - let's create a virtual one to follow scanned one
-                    await validateReceiptDuplicates(context, PersAcc, period, organizationsIds, PersonalAcc)
+                }
 
-                    // find acquiring context and routing number from older receipt
-                    const billingIntegrationContext = await getById('BillingIntegrationOrganizationContext', lastBillingReceipt.context.id)
-                    /** @type {AcquiringIntegrationContext[]} */
-                    const acquiringContexts = await AcquiringIntegrationContext.getAll(context, {
-                        organization: { id: billingIntegrationContext.organization, deletedAt: null },
-                        status: ACQUIRING_CONTEXT_FINISHED_STATUS,
-                        deletedAt: null,
-                    })
+                /** @type {TCompareQRResolvers} */
+                const resolvers = {
+                    onNoReceipt: async () => {
```

Comment:
you can extract it as local function

Reply:
I've extracted the `payForLastBillingReceipt` function to avoid code duplication. But `onNoReceipt` is used only once. All these functions should be defined within mutation cause they use many local variables, mutate a local variable, and call from utilities.